### PR TITLE
Added way to extend rulesets

### PR DESCRIPTION
**Pull request by yannickl88**
Added way to extend rulesets. This is the first step to allowing extending the rules so you can change the matching criteria.

### CODE REVIEW

File: src/Command/SniffCommand.php [hunk 1]
```diff
@@ -91,4 +91,21 @@ private function getFormatter(string $type, bool $pretty_format): FormatterInter
...
+
+        if (file_exists('csssniff.xml.dist')) {
+            return realpath('csssniff.xml.dist');
+        }
+
+        if (file_exists('csssniff.xml')) {
+            return realpath('csssniff.xml');
+        }
+
+        return 'Hostnet';
```
Comment (linaori) [inline src/Command/SniffCommand.php]: Shall we simply provide a "Default" rather than "Hostnet"?

Comment (yannickl88) [reply src/Command/SniffCommand.php]: I want keep the name Hostnet so I can show you are using the Hostnet CSS standard. If we rename it to "Default" if can be confusing what you are using.

**Review by linaori — APPROVED** [review]

File: src/Command/SniffCommand.php [hunk 2]
```diff
@@ -91,4 +91,21 @@ private function getFormatter(string $type, bool $pretty_format): FormatterInter
...
+    {
+        if (null !== $standard) {
+            return $standard;
+        }
+
+        if (file_exists('csssniff.xml.dist')) {
+            return realpath('csssniff.xml.dist');
+        }
+
+        if (file_exists('csssniff.xml')) {
```
Comment (nicoschoenmaker) [inline src/Command/SniffCommand.php]: Shouldn't the `csssniff.xml` file have preference over the `csssniff.xml.dist` file?

File: src/Standard.php [hunk 3]
```diff
@@ -18,14 +18,36 @@ class Standard
      */
     public static function loadFromXmlFile(string $file): self
     {
-        if (!file_exists($file)) {
-            throw new \InvalidArgumentException(sprintf('Cannot find standards file "%s".', $file));
+        if (file_exists($file)) {
+            $standard_file = $file;
+        } else {
+            // Is it a default standard?
+            $standard_file = __DIR__ . '/Standard/' . $file . '.xml';
```
Comment (nicoschoenmaker) [inline src/Standard.php:25]: Should we verify whether there might be stuff like `../` present in `$file`?

File: src/Command/SniffCommand.php (hunk 2 above)
Comment (hboomsma) [reply src/Command/SniffCommand.php]: I agree, this is the default way for phpunit and phpcodesniffer.

**Review by hboomsma — CHANGES_REQUESTED** [review]
Should we add a DTD, this helps IDE's understand which configuration options are available.

I think the `.xml` should have precedence over the `.xml.dist`.

File: src/Command/SniffCommand.php (hunk 2 above)
Comment (yannickl88) [reply src/Command/SniffCommand.php]: ah I had them confused then. Will swap them.

File: src/Standard.php (hunk 3 above)
Comment (yannickl88) [reply src/Standard.php:25]: does it matter? Those paths are handled fine with `file_get_contents`.

Comment (yannickl88) [reply src/Standard.php:25]: or are you referring to line 25?

Comment (hboomsma) [reply src/Standard.php:25]: Might be a security problem, but it is data provided by the user and not from the web, so I see no bug but a feature.

**Review by hboomsma — APPROVED** [review]

File: src/Standard.php (hunk 3 above)
Comment (yannickl88) [reply src/Standard.php:25]: well line 25 is there when you want to "include" the standard "Hostnet". I could check if file contains `^[a-zA-Z_-0-9]+$`.

Comment (nicoschoenmaker) [reply src/Standard.php:25]: Line 25 indeed.

**Pull request merged**